python/pyspark/ml/util.py

```diff
@@ -61,32 +66,82 @@ def _randomUID(cls):
 
 
 @inherit_doc
-class MLWriter(object):
+class BaseReadWrite(object):
+    """
+    Base class for MLWriter and MLReader. Stores information about the SparkContext
+    and SparkSession.
+
+    .. versionadded:: 2.3.0
+    """
+
+    def __init__(self):
+        self.sparkSession = None
+
+    def context(self, sqlContext):
```

Comment:
Yeah it isn't used I just included the function definition because JavaMLWriter which inherits from MLWriter which inherits from BaseReadWrite uses it. I changed the function body to give a NotImplementedError though.